### PR TITLE
v_3_0_0: unify CloudConfig construction

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: v_3_0_0/cloudconfig.go [hunk 1]
```diff
@@ -21,14 +28,21 @@ type CloudConfig struct {
...
 		config:   "",
-		params:   params,
-		template: template,
+		params:   config.Params,
+		template: config.Template,
 	}
 
-	return newCloudConfig, nil
+	return c, nil
 }
```
Comment (kopiczko) [inline v_3_0_0/cloudconfig.go]: This file changed. Other that that it's moving stuff around.

File: v_3_0_0/cloudconfig.go [hunk 2]
```diff
@@ -21,14 +28,21 @@ type CloudConfig struct {
 	template string
 }
 
-func NewCloudConfig(template string, params Params) (*CloudConfig, error) {
-	newCloudConfig := &CloudConfig{
+func New(config Config) (*CloudConfig, error) {
```
Comment (xh3b4sd) [inline v_3_0_0/cloudconfig.go]: I think the naming was right. `v_3_0_0.New()` is not really descriptive. `v_3_0_0.NewCloudConfig()` is though.

Comment (kopiczko) [reply v_3_0_0/cloudconfig.go]: I thought it'll be aliased anyway to something like k8scloudconfigv3.

Should Config be CloudConfigConfig in that case?

Comment (xh3b4sd) [reply v_3_0_0/cloudconfig.go]: Don't know. As I said somewhere yesterday, I think when we have to alias packages, the package name is just wrong and we should rather that than being "forced" to alias the import.

Comment (xh3b4sd) [reply v_3_0_0/cloudconfig.go]: IMO `cloudconfigv3` as package name would work, but then we miss the semver semnatics, wich might be unnecessary anyway.

Comment (kopiczko) [reply v_3_0_0/cloudconfig.go]: I think wen package indicates version we're kind of forced to alias them.

Comment (kopiczko) [reply v_3_0_0/cloudconfig.go]: I changed that but I still think CloudConfigConfig and DefaultCloudConfigConfig is awkwar.

**Pull request merged**